### PR TITLE
Add dropdown for top-level federal agency

### LINKED ISSUE #283: Federal Agencies - Create a Dropdown instead of freeform

### PR DESCRIPTION
**Pull request by neilmb**
# Collect top-level agency for federal requests #

## 🗣 Description ##

For Federal agencies, the authorizing official has to come from the top-level agency and the easiest way to get that agency information is to ask the applicant. For federal agencies, we add a field to the application where they can select the top-level agency.

Screen Shot 2022-12-06 at 15 05 26

The extremely long list of top-level agencies comes from GSA's Qualtrics form options. We can easily edit it to add or subtract agencies as we wish.

The user cost of the very long select box and the countervailing advantages this provides for analysts to have this information on the application form should be checked out with users in our testing. @boonerang  

## 💭 Motivation and context ##

Closes #283.

### CODE REVIEW

**Review by SSPJ — APPROVED** [review]
LGTM 👍 Thanks for adding this, Neil!

The Basics
- [x] I read the code
- [x] I ran the code
 
Documentation
- N/A README (or other documentation) is correct and updated
- [x] all new functions and methods are commented
    - [x] using plain language
- N/A licenses for new dependencies are recorded, if any
- N/A licenses for updated dependencies have been checked for accuracy, if any
- N/A new components have been added to the system diagram
 
Security
- N/A security false positives are documented
- [x] data from external sources is cleaned and clearly marked
- [x] significant user actions are logged to a database table
 
Reliability
- [x] good level of error handling
    - N/A interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests

Accessibility
- N/A New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [x] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [x] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [x] All form inputs have explicit labels
    - N/A All relevant images use an img tag
    - N/A All images have alt attributes
    - N/A Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [x] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [x] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [x] Tab order is logical
    - [x] Form instructions are associated with inputs
    - [x] Site doesn’t timeout unexpectedly. Identify elements that may “timeout” and verify that the user can request more time
    - N/A Tables are coded properly. Tables have proper headers and column attributes
    - [x] Headings are nested properly. Heading elements are nested in a logical way
    - N/A Frames are named. All frames have a name element
    - N/A Flashing elements are compliant. Elements that flash on screen do so at a rate of less than 3 Hz
    - [x] Language is set. The language for the page is set
    - N/A The language for sections on the page that differ from the site language are set
    - [x] CSS is not required to use the page. The page makes sense with or without CSS
        image
        🎉 
    - [x] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [x] Page titles are descriptive
    - N/A Required plugins are linked on the page
 
Device Matrix
- [x] firefox
    - [x] web pages render correctly
    - [x] interactive functionality is operative
- [x] chrome/chromium/edge
    - [x] web pages render correctly
    - [x] interactive functionality is operative
- ❌  safari - still broken
- [x] web page is readable and usable 
    - [x] at 480px
    - [x] at 1024px

**Comment by michelle-rago** [comment]
@neilmb Are we planning to implement a sub-agency menu ([as in the 10x example](https://feedback.gsa.gov/jfe/form/SV_1Im8dTPnjnV3HpP))?

**Comment by neilmb** [comment]
> Are we planning to implement a sub-agency menu (as in the 10x example)?

We are not. If the purpose of this information is to help analysts find the right Authorizing Official, then we don't actually want/need to locate them with their subcomponent of agencies. At least that's what I thought that I heard from talking with @loganmeetsworld. @h-m-f-t would know best, though.

File: src/registrar/forms/application_wizard.py [hunk 1]
```diff
@@ -69,6 +71,154 @@ class OrganizationElectionForm(RegistrarForm):
...
+                "Institute of Peace",
+                "Inter-American Foundation",
+                "International Boundary and Water Commission: United States and Mexico",
+                "International Boundary Commission:  United States and Canada",
+                "International Joint Commission:  United States and Canada",
+                "James Madison Memorial Fellowship Foundation",
+                "Japan-United States Friendship Commission",
+                "John F. Kennedy Center for the Performing Arts",
+                "Legal Services Corporation",
+                "Legislative Branch",
```
[the remark this answers is not in the document]
Comment (h-m-f-t) [reply src/registrar/forms/application_wizard.py:156]: ...I have a suspicion that the original source for this data was GSA's .gov registrar, which did exactly this, combining all of the First Branch into "Legislative Branch".

**Comment by h-m-f-t** [comment]
Agree, I don't think we need to collect sub-agency. 

"_top-level_ federal agency" feels off,  but it's fine for now.

**Comment by michelle-rago** [comment]
Okay. My understanding was that the purpose of the drop-down was to get consistency for org names and to minimize the amount of times that analysts have to correct names. If so, it seems like we'd want to avoid "FEMA" vs "Federal Emergency Management Agency," for example.

**Comment by michelle-rago** [comment]
Other suggestions: 
- I think "Top level federal agency" can just be "Federal agency"
- "Organization Name" should be "Organization name" (to comply with sentence case style. Not sure how picky to get at this point!)
- Can the state drop-down not show "Alabama" by default? Can it just be blank? @igorkorenfeld Do you have thoughts about this?

**Comment by igorkorenfeld** [comment]
Re: Dropdowns 
If possible would like the initial state to read "--Select--" (which is what USWDS has as a default). Blank would be a good 2nd alternative

Re: Agency name
[michelle-rago](https://github.com/michelle-rago) you are right we did want to minimize the issue of differing inputs. But we didn't include subagency because we didn't care too much about the intermediary levels of the organization. Just the top level and the org that's applying (e.g. want to capture "GSA" and "18F" but not "FAS" or "TTS"). The FEMA example is a good case of when we might care about and want to standardize both the top-level org and the registrant's immediate org; assuming analysts want to group them at the org level and no the top agency level (e.g. want to collect all FEMA sites, but not every DHS site). 

In either case I do think we need an "Other" option in the list and allow text field entry if that is the case.

File: src/registrar/forms/application_wizard.py [hunk 2]
```diff
@@ -389,10 +539,32 @@ def get_template_names(self):
...
+            # they have answered this question
+            if organization_type == DomainApplication.OrganizationChoices.FEDERAL:
+                return True
+            return False
+
     def get_context_data(self, form, **kwargs):
         """Add title information to the context for all steps."""
         context = super().get_context_data(form=form, **kwargs)
         context["form_titles"] = TITLES
+        if self.steps.current == "organization_contact":
```
Comment (SSPJ) [inline src/registrar/forms/application_wizard.py:566]:
```suggestion
        if self.steps.current == Step.ORGANIZATION_CONTACT:
```

Came back to reference this PR for the one I'm working on and saw a **nit** I missed on my first pass.

**Comment by neilmb** [comment]
Merging this with as much of the feedback incorporated as I could, but I think that it still remains to make sure that we are handling "Other" and state selection as desired. @calerubensteingsa maybe add something to our QA backlog about it?

File: src/registrar/tests/test_views.py [hunk 3]
```diff
@@ -549,6 +563,21 @@ def test_application_form_conditional_elections(self):
...
+        election_result = election_page.form.submit()
+        # the post request should return a redirect to the contact
+        # question
+        self.assertEquals(election_result.status_code, 302)
+        self.assertEquals(
+            election_result["Location"], "/register/organization_contact/"
+        )
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        contact_page = election_result.follow()
+        self.assertNotContains(contact_page, "Top level federal agency")
```
Comment (neilmb) [inline src/registrar/tests/test_views.py:579]:
```suggestion
        self.assertNotContains(contact_page, "Federal agency")
```

File: src/registrar/tests/test_views.py [hunk 4]
```diff
@@ -517,6 +518,19 @@ def test_application_form_conditional_federal(self):
...
+        federal_page.form["organization_federal-federal_type"] = "executive"
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        federal_result = federal_page.form.submit()
+        # the post request should return a redirect to the contact
+        # question
+        self.assertEquals(federal_result.status_code, 302)
+        self.assertEquals(federal_result["Location"], "/register/organization_contact/")
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        contact_page = federal_result.follow()
+        self.assertContains(contact_page, "Top level federal agency")
```
Comment (neilmb) [inline src/registrar/tests/test_views.py:532]:
```suggestion
        self.assertContains(contact_page, "Federal agency")
```

**Review by neilmb — COMMENTED** [review]
Tweaking for new label

**Pull request merged**